ui/tests/integration/components/date-dropdown-test.js

```diff
@@ -136,36 +143,39 @@ module('Integration | Component | date-dropdown', function (hooks) {
     </div>
     `);
 
-    const monthDropdown = this.element.querySelector('[data-test-popup-menu-trigger="month"]');
-    const yearDropdown = this.element.querySelector('[data-test-popup-menu-trigger="year"]');
+    const monthDropdown = find('[data-test-popup-menu-trigger="month"]');
+    const yearDropdown = find('[data-test-popup-menu-trigger="year"]');
 
-    for (let i = 0; i < 5; i++) {
+    // select each year and assert each month is enabled/disabled correctly
+    for (let yearIdx = 0; yearIdx < 5; yearIdx++) {
       await click(yearDropdown);
-      const dropdownListYears = this.element.querySelectorAll('[data-test-year-list] button');
-      await click(dropdownListYears[i]);
+      const dropdownListYears = findAll('[data-test-year-list] button');
+      await click(dropdownListYears[yearIdx]);
 
       await click(monthDropdown);
-      const dropdownListMonths = this.element.querySelectorAll('[data-test-month-list] button');
+      const dropdownListMonths = findAll('[data-test-month-list] button');
 
-      if (i === 0) {
+      if (yearIdx === 0) {
+        // current year is selected
         for (const [monthIndex, month] of dropdownListMonths.entries()) {
-          if (monthIndex < CURRENT_MONTH) {
+          if (monthIndex <= CURRENT_MONTH) {
             assert.false(
               month.disabled,
-              `${ARRAY_OF_MONTHS[monthIndex]} ${dropdownListYears[i].innerText.trim()} valid`
+              `${ARRAY_OF_MONTHS[monthIndex]} ${dropdownListYears[yearIdx].innerText.trim()} enabled`
             );
           } else {
             assert.true(
               month.disabled,
-              `${ARRAY_OF_MONTHS[monthIndex]} ${dropdownListYears[i].innerText.trim()} disabled`
+              `${ARRAY_OF_MONTHS[monthIndex]} ${dropdownListYears[yearIdx].innerText.trim()} disabled`
             );
           }
         }
       } else {
+        // past year is selected
         for (const [monthIndex, month] of dropdownListMonths.entries()) {
           assert.false(
             month.disabled,
-            `${ARRAY_OF_MONTHS[monthIndex]} ${dropdownListYears[i].innerText.trim()} valid`
+            `${ARRAY_OF_MONTHS[monthIndex]} ${dropdownListYears[yearIdx].innerText.trim()} enabled`
           );
         }
```

Comment:
Here for example you could remove the first year from that array and move the past year assertions to a specific test. Inside the `yearIdx === 0` block you could do something like this to change the assertion from false to true:
```js
const assertMethod = monthIndex <= CURRENT_MONTH ? 'false' : 'true';
assert[assertMethod](
  month.disabled,
  message
);
```

Reply:
I didn't even think to do `assert[method]` - brilliant!